corehq/apps/user_importer/importer.py

```diff
@@ -338,40 +338,54 @@ def create_or_update_web_user_invite(email, domain, role_qualified_id, upload_us
     """
     from corehq.apps.users.views import InviteWebUserView
 
-    primary_location = SQLLocation.by_location_id(primary_location_id)
     invite_fields = {
         'invited_by': upload_user.user_id,
         'invited_on': datetime.utcnow(),
         'tableau_role': tableau_role,
         'tableau_group_ids': tableau_group_ids,
-        'primary_location': primary_location,
         'role': role_qualified_id,
         'profile': profile,
     }
+
+    # It's not clear why only these two were chosen to be recorded as changed values
+    changed_values = {
+        'role_name': role_qualified_id,
+        'profile': profile
+    }
+
+    if primary_location_id is not None:
```

Comment:
Note: Similar change done for primary location.

Additionally, adapt the function to `None` values for assigned locations and primary location.